allennlp/models/semantic_role_labeller.py

```diff
@@ -0,0 +1,191 @@
+from typing import Dict, Any
+
+import torch
+from torch.nn.modules.linear import Linear
+import torch.nn.functional as F
+
+from allennlp.common import Params
+from allennlp.common.tensor import arrays_to_variables, viterbi_decode
+from allennlp.data import Vocabulary
+from allennlp.data.fields import IndexField, TextField
+from allennlp.data import Instance
+from allennlp.modules import Seq2SeqEncoder, TimeDistributed, TextFieldEmbedder
+from allennlp.models.model import Model
+
+
+class SemanticRoleLabeller(Model):
+    """
+    This model performs semantic role labelling using BIO tags using Propbank semantic roles.
+    Specifically, it is an implmentation of `Deep Semantic Role Labelling - What works
```

Comment:
Oh right, I hadn't encountered that one before. No problem changing it, just genuinely couldn't see what you meant!